### PR TITLE
feat(cat): add manifest.yaml spec support

### PR DESCRIPTION
**Pull request by bnchrch**
### TL;DR

Add support for using `manifest.yaml` as a spec input in connector-acceptance-test.

```yaml
# See [Connector Acceptance Tests](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference)
# for more information about how to configure these tests
connector_image: airbyte/source-pokeapi-manifest:dev
acceptance_tests:
  spec:
    tests:
      - spec_path: "manifest.yaml"
  connection:
    bypass_reason: "This is a builder contribution, and we do not have secrets at this time"
  discovery:
    bypass_reason: "This is a builder contribution, and we do not have secrets at this time"
  basic_read:
    bypass_reason: "This is a builder contribution, and we do not have secrets at this time"
  incremental:
    bypass_reason: "This is a builder contribution, and we do not have secrets at this time"
  full_refresh:
    bypass_reason: "This is a builder contribution, and we do not have secrets at this time"
```

### What changed?
- Updated `connector_config_fixture` to handle FileNotFoundError and return `None` if the config file is not found.
- Updated `connector_spec_fixture` to parse manifest spec if `manifest.yaml` is provided.
- Added `is_manifest_file` and `parse_manifest_spec` utilities to handle manifest files.
- Updated test cases to skip if config is not provided.
- Updated dependencies and lock files to fix two issues:
  -  python 3.12 doesnt compile due to a breaking change with setuptools
  - a library was incorrectly pinned

### CODE REVIEW

**Comment by bnchrch** [comment]
* **#39907** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39907?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#42879** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42879?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bnchrch and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by clnoll — COMMENTED** [review]
Looks great @bnchrch! Just a couple of small comments and a question - will we have to skip other tests if `connector_config` is `None`, or are we already skipping them for some reason? (If the former, I think we want a different approach than adding control flow everywhere, but if it's the latter I'm good with what you've got here.)

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py [hunk 1]
```diff
@@ -135,8 +135,11 @@ def skip_oauth_default_method_test_fixture(self, inputs: SpecTestConfig):
             pytest.skip(f"Skipping OAuth is default method test: {inputs.auth_default_method.bypass_reason}")
         return False
 
-    def test_config_match_spec(self, actual_connector_spec: ConnectorSpecification, connector_config: SecretDict):
+    def test_config_match_spec(self, actual_connector_spec: ConnectorSpecification, connector_config: Optional[SecretDict]):
         """Check that config matches the actual schema from the spec call"""
+        if not connector_config:
+            pytest.skip("Config is not provided")
```
[the remark this answers is not in the document]
Comment (bnchrch) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:141]: Great callout!

Ok so I had the thought of the decorator.

But the only test in `TestSpec` that used config was this one.

and I dont think have any other tests/airbyte protocol command where config can be optional.

Though, I could make the skip more explicit for the calling connectors and add a bypass for this test instead.

@clnoll wdyt?

[the remark this answers is not in the document]
Comment (clnoll) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:141]: That makes sense to me, and I see now that the plan is to bypass the other tests in the `acceptance-test-config.yaml`. So I'm good with your approach.

**Review by clnoll — APPROVED** [review]
